IntroToCSharp/Shared/Service/UserService.cs

```diff
@@ -216,8 +216,16 @@ public bool TryGetTask(string path, out DataReference<bool> reference)
     public bool GiveXPAndGold(int xpToGive, int goldToGive)
     {
         if (!_userData.IsLoggedIn) return false;
-        UserStats newStats = new(_userStats.XP + xpToGive, _userStats.Gold + goldToGive);
-        _userData.UserStatsRef?.Set(newStats);
+        if(goldToGive > 0)
+        {
+            UserStats newStats = new(_userStats.XP + xpToGive, _userStats.GoldAcquired + goldToGive, _userStats.GoldSpent);
+            _userData.UserStatsRef?.Set(newStats);
+        }
+        else if(goldToGive < 0)
+        {
+            UserStats newStats = new(_userStats.XP + xpToGive, _userStats.GoldAcquired, _userStats.GoldSpent + (-goldToGive));
+            _userData.UserStatsRef?.Set(newStats);
+        }
```

Comment:
Should be able to eliminate the conditional / repeat code and do something like this:

```csharp
int goldAcquired = _userStats.GoldAcquired + Math.Max(0, goldToGive);
int goldSpent = _userStats.GoldAcquired + Math.Min(0, goldToGive);
```